### PR TITLE
[SPARK-9208][SQL] Remove variant of DataFrame string functions that accept column names.

### PR DESCRIPTION
**Pull request by rxin**
It can be ambiguous whether that is a string literal or a column name.

cc @marmbrus

### CODE REVIEW

**Comment by rxin** [comment]
cc @chenghao-intel

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -1120,9 +1094,7 @@ object functions {
    * @since 1.5.0
    */
   @scala.annotation.varargs
-  def greatest(columnName: String, columnNames: String*): Column = if (columnNames.isEmpty) {
-    sys.error("GREATEST takes at least 2 parameters")
-  } else {
+  def greatest(columnName: String, columnNames: String*): Column = {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: this should be also removed?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -1247,9 +1201,7 @@ object functions {
    * @since 1.5.0
    */
   @scala.annotation.varargs
-  def least(columnName: String, columnNames: String*): Column = if (columnNames.isEmpty) {
-    sys.error("LEAST takes at least 2 parameters")
-  } else {
+  def least(columnName: String, columnNames: String*): Column = {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: this should be also removed?

Comment (chenghao-intel) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Nit: Should we remove the `columnName` version API for both `least` and `greatest`?

**Comment by cloud-fan** [comment]
It's much cleaner now!

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 1 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: this is not a string function?

**Comment by chenghao-intel** [comment]
LGTM in general, the only concern should we add more scaladoc/note, for those string parameters? Just in case people get confused, if it represents a string value or column name.

**Comment by rxin** [comment]
Yup I agree it would be great to improve documentation. I plan to go through all functions during QA to improve documentation.

**Comment by rxin** [comment]
OK I'm going to merge this. I will think about whether we should keep greatest/least.

**Pull request closed without merging**